### PR TITLE
Get latest develop changes into master

### PR DESCRIPTION
**Pull request by monsieurleberre**

### CODE REVIEW

File: src/IPPN.proto [hunk 1]
```diff
@@ -48,20 +32,19 @@ message PeerProtocol {
...
-        bytes correlationID = 1;
-        repeated bytes peerID = 2;
+        bytes correlationId = 1;
+        repeated bytes peerIds = 2;
     }
 
-    message ChallengeRequest {
-        sint64 nonce = 1;
+    message PingRequest {
+        bytes nonce = 1;
```
Comment (nshCore) [inline src/IPPN.proto]:
```suggestion
        bytes correlationId = 1;
```

File: src/IPPN.proto [hunk 2]
```diff
@@ -48,20 +32,19 @@ message PeerProtocol {
...
     }
 
-    message ChallengeResponse {
-        string signedNonce = 1;
-        string publicKey = 2;
+    message PingResponse {
+        bytes nonce = 1;
```
Comment (nshCore) [inline src/IPPN.proto]:
```suggestion
        bytes correlationId = 1;
```

File: src/Transaction.proto [hunk 3]
```diff
@@ -104,19 +106,19 @@ message EntryRangeProof {
...
+    bytes BPrime0 = 11;
     bytes t = 12;
 } 
 
 /**
  * CoinbaseEntry
  *
  * Version: For now only one verison. Workers would be paid in plain tx.
- * Amount: max 8 bytes (positive or negative).
+ * Amount: max 8 bytes (always positive).
```
Comment (nshCore) [inline src/Transaction.proto]: @PaulineBernat review this please.

**Review by nshCore — DISMISSED** [review]
would call the nonce in ping correlationId apart from that all good

**Comment by PaulineBernat** [comment]
Agree with James comment

**Review by franssl — DISMISSED** [review]
Looks good

**Review by PaulineBernat — APPROVED** [review]
approved

**Review by nshCore — APPROVED** [review]
:+1:

**Review by franssl — APPROVED** [review]

**Comment by nshCore** [comment]
Thanks all

**Pull request merged**